### PR TITLE
PSR-118 Added why no NINO page

### PR DESCRIPTION
**Pull request by westwater**

### CODE REVIEW

**Review by kurtlogan — COMMENTED** [review]
You will need to implement cleanup method for the `NationalInsuranceNumberPage` so if the user changes that answer then it will remove the unused value from the cache

File: app/pages/NationalInsuranceNumberPage.scala [hunk 1]
```diff
@@ -19,10 +19,20 @@ package pages
...
+import scala.util.Try
 
 case class NationalInsuranceNumberPage(srn: Srn, index: Max99) extends QuestionPage[Boolean] {
 
   override def path: JsPath = JsPath \ toString
 
   override def toString: String = "nationalInsuranceNumber"
+
+  override def cleanup(value: Option[Boolean], userAnswers: UserAnswers): Try[UserAnswers] =
+    value.fold(Try(userAnswers))(if (_) {
```
Comment (kurtlogan) [inline app/pages/NationalInsuranceNumberPage.scala:33]: If the value is `None` it means the answer hs been removed so this case should remove both answers, something like 
```
value match {
  case Some(true)  => userAnswers.remove(NoNinoPage(srn, index))
  case Some(false) => userAnswers.remove(MemberDetailsNinoPage(srn, index))
  case None             => userAnswers.remove(NoNinoPage(srn, index))
                                     .flatMap(_.remove(MemberDetailsNinoPage(srn, index))
}
```

**Review by kurtlogan — CHANGES_REQUESTED** [review]

File: app/pages/NationalInsuranceNumberPage.scala (hunk 1 above)
Comment (kurtlogan) [reply app/pages/NationalInsuranceNumberPage.scala:33]: Also need some tests for this 😉

Comment (westwater) [reply app/pages/NationalInsuranceNumberPage.scala:33]: Added tests here https://github.com/hmrc/pension-scheme-return-frontend/blob/98331bf2747583df660961f6aed069355e02f4a3/test/pages/NationalInsuranceNumberPageSpec.scala#L35-L50 but good point around the None 👍

**Review by kurtlogan — APPROVED** [review]

**Pull request merged**